### PR TITLE
Add Sentry for frontend JS

### PR DESCRIPTION
**Pull request by di**
This PR adds Sentry error collection for our frontend JS.

It requires setting the `SENTRY_FRONTEND_DSN` environment variable to a valid Sentry DSN.

### CODE REVIEW

File: warehouse/templates/base.html [hunk 1]
```diff
@@ -78,6 +78,14 @@
 
     <link rel="search" type="application/opensearchdescription+xml" title="PyPI" href="{{ request.route_path('opensearch.xml') }}">
 
+    {# This needs to be included first (in order to capture errors when
+     # executing other JS scripts. It also can't be async, because then order
+     # of execution is not guaranteed, and it can't be deferred, because most
+     # of our JS would then execute before it, and we might miss exceptions.
+     #}
+    {% if request.registry.settings.get("sentry.frontend_dsn") -%}
+    <script src="https://cdn.ravenjs.com/3.26.2/raven.min.js" crossorigin="anonymous"></script>
```
Comment (ewdurbin) [inline warehouse/templates/base.html]: can we add a [`integrity`](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) attribute here with the expected checksum? this is a lot more feasible with a versioned url, unlike our Google Analytics and Fastly Insights loads.

**Comment by ewdurbin** [comment]
kinda gross looking, but they do appear to have a solution for async loading documented: https://docs.sentry.io/clients/javascript/install/#async-loading

**Comment by di** [comment]
That would require either `script-src: 'unsafe-inline'` or a bit of rearchitecting of how we serve our JS bundle. The script is pretty small, so I'm not too concerned about how long it will block the DOM.

**Comment by ewdurbin** [comment]
Sounds good to me :)

**Comment by ewdurbin** [comment]
`SENTRY_FRONTEND_DSN` for test.pypi.org and pypi.org! let's roll!

**Pull request merged**